ColumnView

TyJOrtiz opened the pull request:
Added a columnViewBrowser.

Very barebones, only the "Layout Mode" Context menu item works.

yaichenbaum commented on Files/Views/LayoutModes/GenericFileBrowser.xaml:
```diff
@@ -98,11 +98,19 @@
                 <MenuFlyoutItem
-                    x:Uid="StatusBarControlGridViewSmall"
-                    Command="{x:Bind AppSettings.ToggleLayoutModeGridViewSmall}"
-                    Text="Grid View (Small)">
```
It looks like you removed this layout option by mistake.

TyJOrtiz replied:
I fixed that issue

duke7553 commented on Files/Interacts/Interaction.cs:
```diff
@@ -613,6 +613,30 @@ private async void OpenSelectedItems(bool displayApplicationPicker)
+        }
+
+        private async void SetDirectory(string path)
+        {
```
This method seems kind of redundant and probably doesn't belong in Interaction.cs. Does using/improving upon what exists in SetWorkingDirectoryAsync() solve the problem?

duke7553 commented on Files/Views/ModernShellPage.xaml.cs:
```diff
@@ -255,10 +255,11 @@ private void SidebarControl_SidebarItemInvoked(object sender, Controls.SidebarIt
             NavigationToolbar.PathControlDisplayText = FilesystemViewModel.WorkingDirectory;
         }
-
+        public static event EventHandler ReloadColumnView;
```
Why would we need this event? BaseLayout-derived layout modes should load the new content upon being navigated to without the need for something like this in the shell page. 🙂

duke7553 replied:
@TyJOrtiz Why didn't the OnNavigatedTo event work out? You should be able to override it iirc

TyJOrtiz replied:
Ok, I must've done something wrong there. I added the override and it works

duke7553 replied:
Excellent!

TyJOrtiz reviewed (commented):
@yaichenbaum I fixed that issue in a new commit

d2dyno1 commented:
Bugs I've managed to spot:
- If double clicked a directory, two columns would open with the same directory
- "Continue where you left off" does not work with column view

TyJOrtiz commented:
> @TyJOrtiz Good job on your PR!
> I've compiled a list of things for this PR.
> There are some smol bugs I've managed to spot 🐞 :
> (This list also includes things that need? to be done)
> 
> * If double clicked a directory, two (or more) columns would open with the same directory

Fixed

> * "Continue where you left off" does not work with column view

I have no idea how to implement that

> * Context menu is not shown for items, instead it is shown for current directory (which is incorrect) - (I can't reproduce this issue anymore. Guess it fixed it self 🤷‍♂️)
> * Rounded corners on items?

I can fix that

> * When selecting more than one item, progressbar shows up

Fixed

> * When item is selected, the context menu for this specific item is still shown when left-clicked on empty space (Not only in ColumnView)

Trying to fix this 

> * Refreshing directory in ColumnView makes the previous directory the same as the leading one

Fixed

> * Keyboard shortcuts don't work (ctrl+c,v,x,a etc.)
> * Selection Rectangle is not present

Working on it

> * Some items have a border around them:
>   ![image](https://user-images.githubusercontent.com/53011783/100943508-a97f6200-34fd-11eb-87e4-55302385165e.png)
>   .....

Should be fixed

>   and sometimes not:
>   ![image](https://user-images.githubusercontent.com/53011783/100943572-cae04e00-34fd-11eb-9e48-b0a5a69af139.png)
> * FileSystemWatcher is not present
> * No drag & drop
> * Items have click animation whereas listview items doesn't

?

> * Selection options don't work
>   ![image](https://user-images.githubusercontent.com/53011783/100943841-4a6e1d00-34fe-11eb-8c5c-0ef487cdea40.png)

Working on that

yaichenbaum commented on Files/Views/LayoutModes/ColumnViewBrowser.xaml:
```diff
@@ -0,0 +1,2070 @@
+              Background="{TemplateBinding Background}"
+              BorderBrush="{TemplateBinding BorderBrush}"
+              BorderThickness="{TemplateBinding BorderThickness}"
+              CornerRadius="0"
```
```suggestion
```

yaichenbaum commented on Files/Views/LayoutModes/ColumnViewBrowser.xaml:
```diff
@@ -0,0 +1,2070 @@
+              Background="{TemplateBinding Background}"
+              BorderBrush="{TemplateBinding BorderBrush}"
+              BorderThickness="{TemplateBinding BorderThickness}"
+              CornerRadius="0"
```
```suggestion
```

d2dyno1 commented:
> * "Continue where you left off" does not work with column view
@TyJOrtiz
There's a `Path` property in `ITabItem` (- newest commit) that needs to be updated when navigating to a directory.
Continue where you left off uses that property. Do you update it?

d2dyno1 commented:
@TyJOrtiz
There's a `Path` property in `ITabItem` (- newest commit) that needs to be updated when navigating to a directory.
Continue where you left off uses that property. Do you update it?
> > * "Continue where you left off" does not work with column view
> 
> I have no idea how to implement that

TyJOrtiz commented:
> @TyJOrtiz
> There's a `Path` property in `ITabItem` that needs to be updated when navigating to a directory.
> Continue where you left off uses that property. Do you update it?
> 
> > > * "Continue where you left off" does not work with column view
> > 
> > 
> > I have no idea how to implement that

@d2dyno1 Fixed

yaichenbaum commented:
@TyJOrtiz It looks lile the option to rename items isn't working in the column view layout.

TyJOrtiz commented:
@yaichenbaum I'm still working on it, I don't know how to do all that stuff yet. Like I said everything's more complex from when I first started on this so I'm just trying to acquaint myself.

d2dyno1 commented:
> @yaichenbaum I'm still working on it, I don't know how to do all that stuff yet. Like I said everything's more complex from when I first started on this so I'm just trying to acquaint myself.

There's a `IFilesystemHelpers` interface for this

d2dyno1 commented:
> * Refreshing directory in ColumnView removes all previous directories
> * Keyboard shortcuts don't work (ctrl+c,v,x,a etc.)
> * Selection Rectangle is not present
>   ![image](https://user-images.githubusercontent.com/53011783/100943508-a97f6200-34fd-11eb-87e4-55302385165e.png)
>   .....
>   and sometimes not:
>   ![image](https://user-images.githubusercontent.com/53011783/100943572-cae04e00-34fd-11eb-9e48-b0a5a69af139.png)
> * FileSystemWatcher is not present
> * No drag & drop
> * Items should not have click animation
> * Selection options don't work
>   ![image](https://user-images.githubusercontent.com/53011783/100943841-4a6e1d00-34fe-11eb-8c5c-0ef487cdea40.png)
> * Phone contents aren't shown in ColumnView:
>   ![image](https://user-images.githubusercontent.com/53011783/100993982-5e456d80-3556-11eb-8a1d-8c0963cd5f7a.png)
>   .
>   ![image](https://user-images.githubusercontent.com/53011783/100994099-87fe9480-3556-11eb-92fe-d9ebf0e889f7.png)

@TyJOrtiz How's progress going?

TyJOrtiz commented:
> @TyJOrtiz It looks lile the option to rename items isn't working in the column view layout.

@yaichenbaum Renaming file functionality is now there, I will need to make improvements

> Items should not have click animation

@d2dyno1 Fixed

I'll be working on this more throughout the week

d2dyno1 commented:
Hi @TyJOrtiz! Do you need any help with the PR?

lucandrei commented:
I would love to see this live! ❤️

yaichenbaum commented:
@TyJOrtiz Can we get a status update on this pull request? We would be happy to help solve the merge conflicts from `main`.

TyJOrtiz commented:
I'm rewriting the whole ColumnView. I'm still trying to figure out how to make individual instances of the itemviewmodel

mdtauk commented:
> I'm rewriting the whole ColumnView. I'm still trying to figure out how to make individual instances of the itemviewmodel

ColumnView probably wont be compatible with the Split Pane view.  I don't think it would make for a great UI or UX for users.

d2dyno1 commented:
> I'm rewriting the whole ColumnView. I'm still trying to figure out how to make individual instances of the itemviewmodel

@TyJOrtiz don't hesitate to ask for help!

d2dyno1 commented:
![20210128_003223](https://user-images.githubusercontent.com/53011783/106068133-5d692c00-6100-11eb-9409-e2ffefe17adf.jpg)

TyJOrtiz commented:
@d2dyno1 I figured out how to create new viewmodels so now it's full steam ahead

lucandrei commented:
Let's gooo! Thank you 🙌

bl-ue commented:
Any progress?

TyJOrtiz commented:
I created a pull request a couple weeks ago, what I’m working on now is keyboard shortcuts and trying to fix that MTP issue.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

From: bl-ue<mailto:notifications@github.com>
Sent: Tuesday, February 16, 2021 1:10 PM
To: files-community/Files<mailto:Files@noreply.github.com>
Cc: Tyler Ortiz<mailto:imthewiseguy@outlook.com>; State change<mailto:state_change@noreply.github.com>
Subject: Re: [files-community/Files] ColumnView (#2500)


Any progress?

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ffiles-community%2FFiles%2Fpull%2F2500%23issuecomment-780120057&data=04%7C01%7C%7Cb1f57704ae104f33a86408d8d2bf5a65%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637491066580406631%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=BFnghuL%2BkPY5jLb9%2FON21E1iqsjrCU3KSnZMWiRT9dA%3D&reserved=0>, or unsubscribe<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAGBFXDH6OBFXMAI3YUT7EGTS7LNOBANCNFSM4UISACBA&data=04%7C01%7C%7Cb1f57704ae104f33a86408d8d2bf5a65%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637491066580416588%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=bNuI2eoWBlX8u2Gq0shNmfjGUrnLdTTbdhAPfurGP74%3D&reserved=0>.

d2dyno1 commented:
What if we merge ColumnView as is and enable it for debug only. And improve it with further PRs to have this PR finally out of the way?

TyJOrtiz commented:
That sounds like a good idea.

Sent from my iPhone

On Feb 27, 2021, at 1:48 PM, d2dyno1 <notifications@github.com> wrote:

﻿

What if we merge ColumnView as is and enable it for debug only. And improve it with further PRs to have this PR finally out of the way?

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ffiles-community%2FFiles%2Fpull%2F2500%23issuecomment-787165354&data=04%7C01%7C%7C1a809cc11fe745cd994908d8db696b5e%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637500593101154742%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ELn4cntRSZc2w5%2BwoAFwr0VlJl1jsGTrKfAqZxM00F4%3D&reserved=0>, or unsubscribe<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAGBFXDCCQ3LP5QXKWFT6MILTBFSC3ANCNFSM4UISACBA&data=04%7C01%7C%7C1a809cc11fe745cd994908d8db696b5e%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637500593101154742%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=QJmnS4QhNiTndcTKs66mQEgUTBCHqhgCT%2BYNtpAnYtY%3D&reserved=0>.

d2dyno1 commented:
@TyJOrtiz please resolve conflicts

TyJOrtiz commented:
Will do, I'll have it finished before the night's over

TyJOrtiz commented:
I tried to see whats conflicting and it’s saying “too complex to view in browser”. I have no idea what to do

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

From: d2dyno1<mailto:notifications@github.com>
Sent: Saturday, February 27, 2021 3:23 PM
To: files-community/Files<mailto:Files@noreply.github.com>
Cc: Tyler Ortiz<mailto:imthewiseguy@outlook.com>; Mention<mailto:mention@noreply.github.com>
Subject: Re: [files-community/Files] ColumnView (#2500)


@TyJOrtiz<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FTyJOrtiz&data=04%7C01%7C%7C3f4fa5ee38714dddb52d08d8db76a31c%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637500649919185353%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=C5SMRujg8tnq6qz8kCj0SdyZpBVCYlObq7tbcHE9q50%3D&reserved=0> please resolve conflicts

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ffiles-community%2FFiles%2Fpull%2F2500%23issuecomment-787203044&data=04%7C01%7C%7C3f4fa5ee38714dddb52d08d8db76a31c%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637500649919185353%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=FR%2Bsrnn3fiBKTOQaZCtQMhI20xuBiXX58SfzXZDmry0%3D&reserved=0>, or unsubscribe<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAGBFXDGPMD4SN5VY6MLK5ZLTBF5FTANCNFSM4UISACBA&data=04%7C01%7C%7C3f4fa5ee38714dddb52d08d8db76a31c%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637500649919195308%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=N4GZKeWLdQZpffyJLCiv0E0nFjpiUGsRsgJS%2B63g4E8%3D&reserved=0>.

d2dyno1 commented:
@TyJOrtiz in VS, click "Commit and Sync"

d2dyno1 commented:
if that doesn't work,
open cmd in files repo dir:

1 
```
git fetch origin master
```
2
```
git merge origin master
```
If that doesn't work, do:
 
1
```
git remote add upstream https://github.com/files-community/Files.git
```
2
```
git fetch upstream
```
3
```
git merge upstream/master
```

Conflicts will be shown in VS to resolve

TyJOrtiz commented:
I finally merged the main to my fork, I have a question before I freate a pull request...

The context menu no longer shows the Windows Explorer-like context menu (for any view), now its just a basic context menu 
![image](https://user-images.githubusercontent.com/25320332/109414443-ce05a180-7967-11eb-8ccd-b355082ce479.png)

Was that intentional?

gave92 commented:
> The context menu no longer shows the Windows Explorer-like context menu (for any view), now its just a basic context menu
Was that intentional?

Nooope, we need to investigate why this is happening :)

yaichenbaum commented:
@gave92 I have this issue with `main` as well.

bl-ue commented:
Hey, why didn't this get merged? 😢

yaichenbaum commented:
@bl-ue There is a new PR #4176.

bl-ue commented:
Ah, I see. Thank you, @yaichenbaum!
